lib/AST/ASTDemangler.cpp

```diff
@@ -1363,19 +1487,24 @@ ASTBuilder::findDeclContext(NodePointer node) {
 
     auto child = node->getFirstChild();
     if (child->getKind() == Node::Kind::Module) {
-      auto potentialModules = findPotentialModules(child);
+      ModuleDecl *scratch;
+      auto potentialModules = findPotentialModules(child, scratch);
       if (potentialModules.empty())
         return nullptr;
 
-      for (auto *module : potentialModules)
-        if (auto typeDecl = findTypeDecl(module, getIdentifier(name),
+      // FIXME: Looping over potential modules in reverse is a hack to check the
+      //        current main module before checking other ModuleDecls in the
+      //        NameToModules map to avoid an issue where a DeclContext exists
+      //        but doesn't include all desired declarations.
+      for (auto *module : reverse(potentialModules))
```

Comment:
As discussed, I think we should only be considering loaded modules here. Actually thinking about it further, I suspect we could probably just fix this in `addLoadedModule`, if there's already an existing module in `LoadedModules` we should remove it from `NameToModules` too.

Reply:
I updated it and it works fine now, thanks for the suggestion 🙏🏼